src/Controller/Webhook.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Drupal\commerce_ginger\Controller;
+
+use Drupal\commerce\Response\NeedsRedirectException;
+use Drupal\commerce_ginger\Redefiners\BuildersRedefiner;
+use Drupal\commerce_ginger\Builders\OrderBuilder;
+use Drupal\commerce_ginger\Builders\ClientBuilder;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\commerce_ginger\Bankconfigs\Bankconfig;
+use GingerPluginSdk\Client;
+
+class Webhook
+{
+  use StringTranslationTrait;
+
+  /**
+   * @var OrderBuilder
+   */
+  private $orderBuilder;
+  /**
+   * @var Bankconfig
+   */
+  private $bankconfig;
+  /**
+   * @var ClientBuilder
+   */
+  private $clientBuilder;
+  /**
+   * @var Client
+   */
+  private $client;
+
+  public function __construct()
+  {
+    $this->orderBuilder = new OrderBuilder();
+    $this->clientBuilder = new ClientBuilder();
+    $this->bankconfig = new Bankconfig();
+    $this->client = $this->clientBuilder->getClient();
+  }
+
+  public function processWebhook(array $webhookData, $entityTypeManager)
+  {
+    $orderId = filter_var($webhookData['order_id'],FILTER_SANITIZE_STRING);
+    $projectId = filter_var($webhookData['project_id'],FILTER_SANITIZE_STRING);
+
+    $payment = $this->orderBuilder->getOrderPaymentByTransactionId($orderId, $entityTypeManager);
```

Comment:
I don't think the logic of orderBuilder, by sense, should provide such methods. This method should be in some kind of operations controller, like Client or App